provider.go

```diff
@@ -81,6 +83,24 @@ func validate(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	rekorClient, err := rekor.NewClient(defaultRekorURL)
+	if err != nil {
+		sendResponse(nil, fmt.Sprintf("creating Rekor client: %v", err), w)
+		return
+	}
+
+	fulcioRoots, err := fulcio.GetRoots()
+	if err != nil {
+		sendResponse(nil, fmt.Sprintf("getting root certs: %v", err), w)
+		return
+	}
+
+	fulcioIntermediates, err := fulcio.GetIntermediates()
+	if err != nil {
+		sendResponse(nil, fmt.Sprintf("getting Fulcio intermediates: %v", err), w)
+		return
+	}
```

Comment:
Can you please move this initialization logic to `main()` function? By doing so, we can avoid to init a new rekor client on every request. So it should be singleton instance. Maybe you can move the required variables to global var: `RekorClient`, `RootCerts`, `IntermediateCerts`.

Reply:
Good point, yeah saw that in your other PR. I will do it, thanks!

Reply:
Done, let me know if there is anything else, thanks!